src/metadataGeneration/customAttribute.ts

```diff
@@ -0,0 +1,67 @@
+import * as ts from 'typescript';
+import { getInitializerValue } from './initializer-value';
+import { Tsoa } from './tsoa';
+
+const getArgumentValue = (argument: ts.Expression) => {
+  switch (argument.kind as ts.SyntaxKind) {
+    case ts.SyntaxKind.StringLiteral: {
+      const argumentValue = (argument as any).text;
+
+      return argumentValue;
+    }
+
+    case ts.SyntaxKind.ObjectLiteralExpression: {
+      const argumentValueProperties = (argument as any).properties;
+
+      const argumentValue = {};
+
+      for (const property of argumentValueProperties) {
+        const name = property.name.text;
+        const scopes = getInitializerValue(property.initializer);
+        argumentValue[name] = scopes;
+      }
+
+      return argumentValue;
+    }
+
+    case ts.SyntaxKind.ArrayLiteralExpression: {
+      const elements = (argument as any).elements;
+
+      const argumentValue = elements.map((element) => getArgumentValue(element));
+
+      return argumentValue;
+    }
+
+    default: {
+      throw new Error('Custom Attribute values must be of type: string, object, or array');
+    }
+  }
+};
+
+export function getCustomAttributes(decorators: ts.Identifier[]): Tsoa.CustomAttribute[] {
+  const customAttributes: Tsoa.CustomAttribute[] = decorators.map((customAttributeDecorator: ts.Identifier) => {
+    const expression = customAttributeDecorator.parent as ts.CallExpression;
+
+    const [decoratorKeyArg, decoratorValueArg] = expression.arguments;
```

Comment:
If you have the suggested check there, this is safe to do without any casting: 
```suggestion
    const [decoratorKeyArg, decoratorValueArg] = customAttributeDecorator.parent.arguments;
```